### PR TITLE
ui: Redesign - Node service instances tab

### PR DESCRIPTION
**Pull request by kaxcode**
- Implemented the ConsulServiceInstancesList component to the Node show page
- Updated the ConsulServiceInstacesList component to work for Nodes and Services
- Created ConsulInstaceChecks component to refactor code and make the template file less busy
- Updates Node test to work with ConsulServiceInstancesList 

Screen Shot 2020-06-29 at 12 45 50 PM

### CODE REVIEW

File: ui-v2/app/components/consul-instance-checks/index.hbs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{{#if (gt items.length 0)}}
+{{#let (service/instance-checks items) as |itemCheck| }}
```
Comment (johncowen) [inline ui-v2/app/components/consul-instance-checks/index.hbs:2]: If you don't see us using it anywhere else, and you wanted to, you could potentially move this helper code into the component, completely up to you tho.

File: ui-v2/app/router.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ export const routes = {
           _options: { path: '/health-checks' },
         },
         services: {
-          _options: { path: '/services' },
+          _options: { path: '/service-instances' },
```
Comment (johncowen) [inline ui-v2/app/router.js:70]: Do we mind that we are changing a URL here? 🤔

File: ui-v2/app/components/consul-instance-checks/index.hbs (hunk 1 above)
Comment (kaxcode) [reply ui-v2/app/components/consul-instance-checks/index.hbs:2]: I don't know how to add all that logic to the component. I'll leave it as it is for now.

File: ui-v2/app/components/consul-node-service-list/index.hbs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        {{/if}}
+      </li>
+    {{#if (gt item.Service.Tags.length 0)}}
+      <li>
+        <TagList @items={{item.Service.Tags}}/>
+      </li>
+    {{/if}}
+    </ul>
+  </ListCollection>
+{{/if}}
```
[the remark this answers is not in the document]
Comment (kaxcode) [reply ui-v2/app/components/consul-node-service-list/index.hbs:47]: Deleted

File: ui-v2/app/router.js (hunk 2 above)
Comment (kaxcode) [reply ui-v2/app/router.js:70]: I'll leave it as '/service-instances'

**Comment by kaxcode** [comment]
I decided to leave the old SearchBar. I need to discuss with designers if the Search/Filter will change again for this page. Will make separate ticket afterwards.

File: ui-v2/app/components/consul-instance-checks/index.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import Component from '@ember/component';
+import { computed, get } from '@ember/object';
+
+export default Component.extend({
+  tagName: '',
+  healthCheck: computed(function() {
```
Comment (johncowen) [inline ui-v2/app/components/consul-instance-checks/index.js:6]: We need to make sure we add any dependency properties as a first argument here, so in this case you'd need to add `items.[]` which will cover any changes to the `items` array.

File: ui-v2/app/templates/dc/nodes/show/services.hbs [hunk 5]
```diff
@@ -1,3 +1,12 @@
+{{#let (selectable-key-values
+    (array "Name:asc" "A to Z")
+    (array "Name:desc" "Z to A")
+    (array "Status:asc" "Unhealthy to Healthy")
+    (array "Status:desc" "Healthy to Unhealthy")
+      selected=sortBy
+  )
+  as |sort|
+}}
```
Comment (johncowen) [inline ui-v2/app/templates/dc/nodes/show/services.hbs:9]: I think you can take this out for the moment also if you aren't sorting yet?

Comment (kaxcode) [reply ui-v2/app/templates/dc/nodes/show/services.hbs]: Oops. Removed.

**Pull request merged**